src/opensubtitlescom/config.py

```diff
@@ -0,0 +1,55 @@
+"""
+This file is part of Opensubtitles API wrapper.
+
+Opensubtitles API is free software: you can redistribute it and/or modify
+it under the terms of the MIT License as published by the Massachusetts
+Institute of Technology.
+
+For full details, please see the LICENSE file located in the root
+directory of this project.
+"""
+import os
+import json
+from typing import Optional
+from pathlib import Path
+
+
+DEFAULT_CONFIG_PATH = "~/.config/opensubtitlescom/cli.json"
+
+
+class Config:
+    """Represents the configuration for the OpenSubtitles CLI."""
+
+    def __init__(self, path: Optional[Path] = None):
+        """Initialize the Config object.
+
+        Args:
+            path (Path): The path to the configuration file.
+
+        Note:
+            If the configuration file exists, the class is initialized with its content.
+            If not, the class creates the necessary directory structure for the configuration file.
+        """
+        self._path: Path = (Path(path or DEFAULT_CONFIG_PATH)).expanduser()
+        self.api_key: Optional[str] = None
+        self.app_name: Optional[str] = None
+        self.app_version: Optional[str] = "0.0.1"
```

Comment:
Putting these into the config object means that they will be saved to the config file - and so if this library or CLI ever increments the version number, users won't get the new version number, because the old version is stored in the file.

(It also means that if multiple different apps use this library, all of the different apps will end up being given the same app name and version)

IMO `Config` should only deal with user-config -- application-config should be part of the application

Reply:
There seems to be a misunderstanding. The `app_version` here doesn't represent the library version; 
instead, it serves as a suffix to the "User-Agent" header.
It's a default value and isn't stored in the config file. 
If a value exists in the config file, it takes precedence over the default.

Reply:
In save(), every field (except those with underscores) from the Config object is written to the config file, so:

- A user runs MyApp v1.0, which calls this code and then calls save() to write the config to disk
- "MyApp v1.0" gets written into the config file
- A user then upgrades to MyApp v2.0, which calls this code
- The app reads "MyApp v1.0" from the config file, and goes to the server claiming to be v1.0
- The user switches to SomeOtherApp v1.2.3
- SomeOtherApp loads this config file, and talks to the server claiming to be MyApp v1.0, because that's what was in the config file

I think this approach is mixing up "library config" and "application config" when those two things have different needs - IMO:
- the library should accept username / password / app name / app version as function parameters (not stored to disk anywhere)
- the application (in this case, the CLI) should accept username / password as parameters (from command line or from config file or wherever), and have app name / app version hard-coded